nex_server.go

```diff
@@ -30,27 +33,28 @@ func startNEXServer() {
 
 	globals.NEXServer.On("Kick", func(packet *nex.PacketV0) {
 		pid := packet.Sender().PID()
-		delete(globals.ConnectedUsers, pid)
 
+		if globals.ConnectedUsers[pid] == nil {
+			return
+		}
+
+		platform := globals.ConnectedUsers[pid].Platform
 		lastOnline := nex.NewDateTime(0)
 		lastOnline.FromTimestamp(time.Now())
 
-		database_wiiu.UpdateUserLastOnlineTime(pid, lastOnline)
-		sendUserWentOfflineWiiUNotifications(packet.Sender())
+		if platform == 1 {
+			database_wiiu.UpdateUserLastOnlineTime(pid, lastOnline)
+			friends_wiiu.SendUserWentOfflineNotifications(packet.Sender())
+		} else if platform == 2 {
+			database_3ds.UpdateUserLastOnlineTime(pid, lastOnline)
+			friends_3ds.SendUserWentOfflineNotificationsGlobally(packet.Sender())
+		}
```

Comment:
These platforms should probably be an enum or global variables for easier reading. Such as `WUP`/`CTR` (since variables can't start with a number for the 3DS)